helix-term/src/commands.rs

```diff
@@ -2148,7 +2162,10 @@ fn buffer_picker(cx: &mut Context) {
             .collect(),
         BufferMeta::format,
         |cx, meta, action| {
-            cx.editor.switch(meta.id, action);
+            if let Some(meta) = meta {
+                cx.editor.switch(meta.id, action);
+            }
+            picker::close_picker()
```

Comment:
I don't quite like this boilerplate, I tried doing a helper function but then it prevents type elision and so I had to put in the type, I think not really that good so I didn't do it and annotate all functions with this.

Reply:
I just make it separate struct since we want to handle a bunch of custom stuff also.